### PR TITLE
Plat UI 2706 govuk frontend v5 0 0 rebase

### PR DESCRIPTION
**Pull request by JoPintoPaul**

### CODE REVIEW

**Comment by ellamdav** [comment]
Remember to update the release date in the CHANGELOG once approved 😅

File: play-frontend-hmrc-play-30/src/main/scala/uk/gov/hmrc/hmrcfrontend/views/config/StandardPhaseBanner.scala
Comment (oscarduignan) [inline play-frontend-hmrc-play-30/src/main/scala/uk/gov/hmrc/hmrcfrontend/views/config/StandardPhaseBanner.scala:54]: should these be Beta and Alpha rather than BETA and ALPHA - the [examples on govuk](https://design-system.service.gov.uk/components/phase-banner/) are just capitalized but not fully uppercased

File: play-frontend-hmrc-play-30/src/main/twirl/uk/gov/hmrc/govukfrontend/views/components/GovukCookieBanner.scala.html
Comment (oscarduignan) [inline play-frontend-hmrc-play-30/src/main/twirl/uk/gov/hmrc/govukfrontend/views/components/GovukCookieBanner.scala.html:24]: damn mind bending twirl `@if(hidden){ hidden}` being "output 'hidden' literally" rather than the value of the hidden variable :oof:

File: play-frontend-hmrc-play-30/src/main/twirl/uk/gov/hmrc/govukfrontend/views/components/GovukCookieBanner.scala.html
Comment (oscarduignan) [inline play-frontend-hmrc-play-30/src/main/twirl/uk/gov/hmrc/govukfrontend/views/components/GovukCookieBanner.scala.html:25]: guessing fixtures would catch if it wasn't right but 
```
@message.role.mapNonEmpty {role => role="@role"}
```
seems really mind bending to me too - in my head I was expecting `{role => role="@role"}` to be scala rather than a twirl block because I read @ in this case as output variable, so you'd need to be building a string - #twirlthings 😵

Comment (oscarduignan) [reply play-frontend-hmrc-play-30/src/main/twirl/uk/gov/hmrc/govukfrontend/views/components/GovukCookieBanner.scala.html:25]: in my head it's difference between interpretting stuff as a twirl statement
```
@if(message.role.nonEmpty) { role="@message.role"}
```
vs inline scala
```
@{if (message.role.nonEmpty) s"""role="$message.role"""" else ""}
```
guess my intuition is getting where it's `@var` a bit wrong - want like a twirl cheat sheet / test harness with table of "this renders as that"

File: play-frontend-hmrc-play-30/src/main/twirl/uk/gov/hmrc/govukfrontend/views/components/GovukCookieBanner.scala.html
Comment (oscarduignan) [inline play-frontend-hmrc-play-30/src/main/twirl/uk/gov/hmrc/govukfrontend/views/components/GovukCookieBanner.scala.html:42]: small thing to see if I missed something but why did we change this to action.href.getOrElse("").isEmpty rather than action.href.exists(_.nonEmpty) seems like more steps where exists would short circuit straight away on None

File: play-frontend-hmrc-play-30/src/test/scala/uk/gov/hmrc/govukfrontend/views/components/GovukPhaseBannerSpec.scala
Comment (oscarduignan) [inline play-frontend-hmrc-play-30/src/test/scala/uk/gov/hmrc/govukfrontend/views/components/GovukPhaseBannerSpec.scala:23]: :chef-kiss:

File: play-frontend-hmrc-play-30/src/test/scala/uk/gov/hmrc/govukfrontend/views/viewmodels/tasklist/TaskListItemSpec.scala
Comment (oscarduignan) [inline play-frontend-hmrc-play-30/src/test/scala/uk/gov/hmrc/govukfrontend/views/viewmodels/tasklist/TaskListItemSpec.scala:1]: Should we just have a TaskListSpec - seems like these just check if serializing works - and since these are all used in the TaskList case class I guess just running tests on that covers all these too - can see other ones have it separately too though - so guess it doesn't really matter 🤷

File: play-frontend-hmrc-play-30/src/test/scala/uk/gov/hmrc/hmrcfrontend/views/components/HmrcFooterSpec.scala
Comment (oscarduignan) [inline play-frontend-hmrc-play-30/src/test/scala/uk/gov/hmrc/hmrcfrontend/views/components/HmrcFooterSpec.scala:37]: small one, but I think we could reduce this to
```suggestion
    implicit val messages = messagesApi.preferred(Seq(Lang(templateParams.language.code)))
```
MessagesSupport trait could also be changed to `messages = mesagesApi.preferred(FakeRequest())` or use the seq in the defaultLangs in that trait rather than a fake request, but 🤷‍♂️

File: play-frontend-hmrc-play-30/src/test/scala/uk/gov/hmrc/hmrcfrontend/views/helpers/hmrcInternalHeadSpec.scala
Comment (oscarduignan) [inline play-frontend-hmrc-play-30/src/test/scala/uk/gov/hmrc/hmrcfrontend/views/helpers/hmrcInternalHeadSpec.scala:88]: Ah, just spotted that html5shiv is gone - I think we were going to keep this for a bit longer because without it the pta/bta/sca account menu breaks in IE8 - think some discussion around it relating to PLATUI-2684 in slack

File: play-frontend-hmrc-play-30/src/main/scala/uk/gov/hmrc/hmrcfrontend/views/config/StandardPhaseBanner.scala
Comment (ellamdav) [reply play-frontend-hmrc-play-30/src/main/scala/uk/gov/hmrc/hmrcfrontend/views/config/StandardPhaseBanner.scala:54]: I think we were just trying to maintain parity with what's there right now - but it feels like the research that led to this change found that Sentence case was more accessible, so maybe we should change them?  Is it maybe a question for DR?

File: play-frontend-hmrc-play-30/src/main/twirl/uk/gov/hmrc/govukfrontend/views/components/GovukCookieBanner.scala.html
Comment (ellamdav) [reply play-frontend-hmrc-play-30/src/main/twirl/uk/gov/hmrc/govukfrontend/views/components/GovukCookieBanner.scala.html:42]: I think we're trying to catch where `href` is `None` or `Some("")`, which using `exists` would probably be `!action.href.exists(_.nonEmpty)` which is arguably more mind-bending 😅 
Broadly speaking though, there are many places where we're having to code defensively purely to workaround the ambiguous JS model (see `Option[Seq[...]]`) and/or our own generative tests that populate those models with illogical inputs - it could be worth looking at some more strict typing here, or `require`ing some preconditions to make these cases go away... or it could just turn out to be yak-shaving 🤷

File: play-frontend-hmrc-play-30/src/test/scala/uk/gov/hmrc/govukfrontend/views/viewmodels/tasklist/TaskListItemSpec.scala
Comment (ellamdav) [reply play-frontend-hmrc-play-30/src/test/scala/uk/gov/hmrc/govukfrontend/views/viewmodels/tasklist/TaskListItemSpec.scala:1]: I guess we just followed the prevailing convention here - the test of the outer class should be enough, but these tests seem pretty quick so 🤷

File: play-frontend-hmrc-play-30/src/main/scala/uk/gov/hmrc/hmrcfrontend/views/config/StandardPhaseBanner.scala
Comment (JoPintoPaul) [reply play-frontend-hmrc-play-30/src/main/scala/uk/gov/hmrc/hmrcfrontend/views/config/StandardPhaseBanner.scala:54]: Let's use sentence case rather than capitalised and put it in the release notes.

File: play-frontend-hmrc-play-30/src/test/scala/uk/gov/hmrc/hmrcfrontend/views/components/HmrcFooterSpec.scala
Comment (ellamdav) [reply play-frontend-hmrc-play-30/src/test/scala/uk/gov/hmrc/hmrcfrontend/views/components/HmrcFooterSpec.scala]: Nice - works a treat 🙌

File: play-frontend-hmrc-play-30/src/main/twirl/uk/gov/hmrc/govukfrontend/views/components/GovukCookieBanner.scala.html
Comment (oscarduignan) [reply play-frontend-hmrc-play-30/src/main/twirl/uk/gov/hmrc/govukfrontend/views/components/GovukCookieBanner.scala.html:42]: ah, I get you, I totally misread the changed logic of it 🤦‍♂️

File: play-frontend-hmrc-play-30/src/main/scala/uk/gov/hmrc/hmrcfrontend/views/config/StandardPhaseBanner.scala
Comment (oscarduignan) [reply play-frontend-hmrc-play-30/src/main/scala/uk/gov/hmrc/hmrcfrontend/views/config/StandardPhaseBanner.scala]: I asked in int-plat-design and Tim has said to use Alpha / Beta 🙌

File: play-frontend-hmrc-play-30/src/test/scala/uk/gov/hmrc/hmrcfrontend/views/helpers/hmrcInternalHeadSpec.scala
Comment (oscarduignan) [reply play-frontend-hmrc-play-30/src/test/scala/uk/gov/hmrc/hmrcfrontend/views/helpers/hmrcInternalHeadSpec.scala:88]: (this was because we're removing it and will tell maintainers of sca-wrapper and they can add it in if they want (or add a wrapping div to avoid needing extra js)

**Review by oscarduignan — APPROVED** [review]

**Review by oscarduignan — APPROVED** [review]

**Pull request merged**
